README.md

```diff
@@ -129,6 +130,53 @@ git/
 Currently, kernels are built using the `bzImage` for x86\_64 or `Image.gz` for
 arm64, and `tar-pkg` targets (see [pkg/kernels/conf.go](pkg/kernels/conf.go)).
 
+### Download kernels
+
+List the available versions
+
+```console
+$ lvh kernels catalog
+bpf-next
+rhel8
+4.9
+4.19
+[...]
+6.3
+6.6
+```
+
+Retrieve the tags for a given version:
+```console
+$ lvh kernels catalog 6.6
+6.6-20240123.120815
+6.6-20240123.175813
+[...]
+6.6-20240404.144247
+6.6-20240408.100959
+6.6-main
+```
+
+See `lvh kernels catalog --help` for more details.
+
+Download the kernel and related artifacts (BTF, modules, etc.)
+
+```console
+$ lvh kernels pull 6.6-main
```

Comment:
I [added the `--ci` flag to pull](https://github.com/cilium/little-vm-helper/compare/8fd3986bcbb1c633436fcc87bcac80383cd44107..8a969ea9cbe6c77bf45c9a210e5e64d3918ded71) so that we have access to the main repo and the ci version at least.

Reply:
I'll have to think it through a bit more, but I was thinking about a workflow something like this:
1. I'm either managing a renovate PR on cilium/cilium that bumps an lvh image, or debugging some issue in a workflow that I would like to test out manually.
2. I see that the cilium/cilium change refers to `quay.io/lvh-images/complexity-test:5.4-20240305.092417`, so I copy+paste this OCI path to my commandline and try to run `lvh run ... --kernel quay.io/lvh-images/complexity-test:5.4-20240305.092417`.

From this perspective, it's nice to be able to copy+paste the exact URI know that you're pulling the same resource. That way I'm not mixing up the wrong kernel image from somewhere just because I'm not familiar with LVH or which repo is being used.

All this I think is just CLI parsing, we shouldn't need to get into the format; I assume that I'm passing the correct format of image.

Obviously if you give it a completely wrong image then this would fail, and I think that's fine. Over time we can make the code more resilient so that it fails loudly when you provide a nonsensical image.

Reply:
(I may/may not have quite the right image / OCI path above but I think you should get a sense for the workflow. I would see a kernel image reference somewhere and copy it into the commandline locally to use).

If you prefer not to follow this, then the other nit I might say is that maybe the flag should be something like `--repo` / `--repository` / `--oci-repo` / `--container-repo` to represent that it's talking about the image repository path.

Reply:
> I see that the cilium/cilium change refers to quay.io/lvh-images/complexity-test:5.4-20240305.092417, so I copy+paste this OCI path to my commandline and try to run lvh run ... --kernel quay.io/lvh-images/complexity-test:5.4-20240305.092417.
>
> From this perspective, it's nice to be able to copy+paste the exact URI know that you're pulling the same resource. That way I'm not mixing up the wrong kernel image from somewhere just because I'm not familiar with LVH or which repo is being used.
>
> All this I think is just CLI parsing, we shouldn't need to get into the format inside the image; I assume that I'm passing the correct format of image.

yes I understand how convenient it would be however two things:
1. This PR does not make `kernel run --kernel <URL>` work, it's a helper around browsing and downloading kernel artifacts. However it should not be too difficult to finish to arrive to this.
2. While being able to retrieve the kernel from `quay.io/lvh-images/complexity-test:5.4-20240305.092417` would be convenient, as in the workflow you showed, the image structure is different. `lvh-images/kernel-images` are storing the kernel binaries in the fs while the complexity, kind or base image are containing a qcow2 image compressed. So it would require extracting the image and then starting a VM (using libguestfs maybe) to copy the file out of the image which is different than treating a `lvh-images/kernel-images`.

I think ultimately, the workflow you presented would be the endgoal UX as you could use any image as kernel so that you just extract the kernel and run it.

But I would say that this PR is slightly different and mostly a wrapper around `lvh-images/kernel-images` which is not a VM image like the others but mostly an archive around the kernel binaries. I don't know if it makes sense to see why it's very linked to this OCI repo and cannot be hot swapped with another.

Reply:
To get to the state that is interesting for you we would require two steps it seems, for this command:
- handle any other VM images we produce as kernel images (if they contain kernels, they might not?)
- wire this command into `lvh run --kernel <...>` while now this is a two steps process.

Reply:
That explanation makes sense to me, and this PR already provides value so we don't need to pull in the scope of that whole proposal here :+1: .